### PR TITLE
Redistribution clean additions

### PR DESCRIPTION
**Pull request by significance**

### CODE REVIEW

**Review by istae — APPROVED** [review]

File: src/PriceOracle.sol [hunk 1]
```diff
@@ -95,7 +96,7 @@ contract PriceOracle is AccessControl {
             // redundancy may not be zero
             require(redundancy > 0, "unexpected zero");
 
-            // maximum extra redundancy is 4
+            // maximum considered extra redundancy is 4
```
Comment (agazso) [inline src/PriceOracle.sol:99]: To me this is a bit strange that the comment here talks about `4` but then the actual code it checks if the redundancy is bigger than 8. I can think any of the following could be an improvement:
- Mentioning in the comment that above a certain value there is no need for more redundancy
- Maybe extract the value 8 to a constant and name it to something that is helpful in this context

File: src/Redistribution.sol [hunk 2]
```diff
@@ -13,36 +13,37 @@ import "./Staking.sol";
...
+    // An eligible user may commit to an _obsfucatedHash_ during the commit phase...
     struct Commit {
         bytes32 overlay;
         address owner;
         uint256 stake;
         bytes32 obfuscatedHash;
         bool revealed;
     }
-    // ...then provide the actual values that are the pre-image of the _obsfucatedHash_
+    // ...then provide the actual values that are the constituents of the pre-image of the _obsfucatedHash_
```
Comment (agazso) [inline src/Redistribution.sol:46]:
```suggestion
    // ...then provide the actual values that are the constituents of the pre-image of the _obfuscatedHash_
```

File: test/PostageStamp.test.ts [hunk 3]
```diff
@@ -53,1184 +73,1241 @@ describe('PostageStamp', function () {
...
-            this.batch.bucketDepth,
-            this.batch.immutable
-          );
-        const stamp = await this.postageStamp.batches(this.batch.id);
-        expect(stamp[3]).to.equal(3 * price + this.expectedNormalisedBalance);
+        expect(await postageStamp.pot()).equal(expectedPot3);
       });
     });
 
+    // describe('when copyBatch creates a batch', function () {
```
Comment (istae) [inline test/PostageStamp.test.ts:645]: I would remove commented code

File: test/PostageStamp.test.ts [hunk 4]
```diff
@@ -53,1184 +73,1241 @@ describe('PostageStamp', function () {
...
-          this.batch.immutable
-        );
-        const batchD = computeBatchId(stamper, nonceD);
+        expect(batch0).not.equal(await postageStamp.firstBatchId());
+        expect(batch1).not.equal(await postageStamp.firstBatchId());
+        expect(batch2).equal(await postageStamp.firstBatchId());
+      });
 
-        expect(await this.postageStamp.pot()).equal(17 * 2 ** this.batch.depth);
+      // todo add this and perhaps other, longer tests
```
Comment (istae) [inline test/PostageStamp.test.ts:473]: same here

File: test/PostageStamp.test.ts [hunk 5]
```diff
@@ -53,1184 +73,1241 @@ describe('PostageStamp', function () {
...
           )
-        ).to.be.revertedWith('normalisedBalance cannot be zero');
-      });
-
-      it('should not return empty batches', async function () {
-        await this.postageStamp.createBatch(
+        ).to.be.revertedWith(errors.createBatch.alreadyExists);
+      });
+
+      // it('should not allow normalized balance to be zero', async function () {
```
Comment (istae) [inline test/PostageStamp.test.ts:326]: here

**Review by istae — APPROVED** [review]

File: test/util/tools.ts [hunk 6]
```diff
@@ -8,24 +8,29 @@ function computeBatchId(sender: string, nonce: string): string {
...
 }
 
-async function getBlockNumber(): Promise<any> {
+async function getBlockNumber(): Promise<number> {
   const blockNumber = await ethers.provider.send('eth_blockNumber', []);
   return parseInt(blockNumber);
 }
 
-async function setPrevRandDAO(randVal: string): Promise<any> {
+async function setPrevRandDAO(randVal: string): Promise<undefined> {
```
Comment (mrekucci) [inline test/util/tools.ts:23]: This function is not used, so it should be removed.

File: test/PostageStamp.test.ts (hunk 3 above)
Comment (mrekucci) [inline test/PostageStamp.test.ts:645]: For what reason are these tests commented/disabled?

File: src/PostageStamp.sol [hunk 7]
```diff
@@ -132,7 +131,7 @@ contract PostageStamp is AccessControl, Pausable {
     ) external whenNotPaused {
         require(_owner != address(0), "owner cannot be the zero address");
         // bucket depth should be non-zero and smaller than the depth
-        require(_bucketDepth != 0 && _bucketDepth < _depth && minimumBatchDepth <= _bucketDepth, "invalid bucket depth");
```
[the remark this answers is not in the document]
Comment (significance) [reply src/PostageStamp.sol:135]: nice yes missed this 🦅

[the remark this answers is not in the document]
Comment (significance) [reply src/PostageStamp.sol:135]: there was a role for it too that needed to go too though 👀 😀

File: src/Redistribution.sol (hunk 2 above)
Comment (significance) [reply src/Redistribution.sol:46]: 🦅

File: src/PriceOracle.sol (hunk 1 above)
Comment (significance) [reply src/PriceOracle.sol:99]: i have extracted the constants and tried to clarify this, hopefully it makes sense now. imo there is a pending task to fix conventions eg. capitalising comments and standardising use of a preceding underscore.

File: src/Redistribution.sol [hunk 8]
```diff
@@ -13,36 +13,37 @@ import "./Staking.sol";
...
+ * When the _reveal_ phase has been concluded, the claim phase can begin. At this point, the truth teller and winner
+ * are already determined. By calling _isWinner_, an applicant node can run the relevant logic to determine if they have
  * been selected as the beneficiary of this round. When calling _claim_, the current pot from the PostageStamp contract
- * is withdrawn and transferred to the beneficiaries address. Nodes that have revealed values that differ from the truth,
+ * is withdrawn and transferred to that beneficiaries address. Nodes that have revealed values that differ from the truth,
  * have their stakes "frozen" for a period of rounds proportional to their reported depth.
  */
 contract Redistribution is AccessControl, Pausable {
-    // An elgible user may commit to an obsfucated hash during the commit phase...
+    // An eligible user may commit to an _obsfucatedHash_ during the commit phase...
```
[the remark this answers is not in the document]
Comment (significance) [reply src/Redistribution.sol:38]: 🔍

File: test/PostageStamp.test.ts (hunk 3 above)
Comment (significance) [reply test/PostageStamp.test.ts:645]: removed for now

**Review by agazso — APPROVED** [review]

File: test/PostageStamp.test.ts (hunk 3 above)
Comment (significance) [reply test/PostageStamp.test.ts]: tracked here https://github.com/ethersphere/storage-incentives/issues/64

Comment (significance) [reply test/PostageStamp.test.ts]: we should rewrite these as part of review of bootstrapping procedure

**Pull request merged**